File: Backend/blocked-users/api/create/service.php
<?php
require_once '../../db/database.php';

function blockUser($id, $email, $reason) {
    $db = new Database();
    $stmt = $db->conn->prepare("INSERT INTO blocked_users (id, email, reason, blocked_at) VALUES (?, ?, ?, ?)");
    
    try {
        $stmt->execute([$id, $email, $reason, date("Y-m-d H:i:s")]);
        return ["message" => "User blocked successfully"];
    } catch (PDOException $e) {
        return ["error" => "Error blocking user"];
    }
}
?>

File: Backend/blocked-users/api/create/service.rb
require_relative "../../db/database"
require "securerandom"

class BlockUserService
  def self.block(email, reason)
    db = Database.connection
    db.execute("INSERT INTO blocked_users (id, email, reason, blocked_at) VALUES (?, ?, ?, ?)", 
               [SecureRandom.uuid, email, reason, Time.now.to_s])
    { message: "User blocked successfully" }
  rescue SQLite3::Exception => e
    { error: "Error blocking user: #{e.message}" }
  end
end
